site/block-diagram/block_diagram.js

```diff
@@ -0,0 +1,165 @@
+/* Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Fetch the latest stats for a block diagram and add them as tooltips.
+ *
+ * @param statsURL URL to the JSON stats report file for this block diagram.
+ * @param blocks   Collection of HTMLElements for the block-diagram's blocks.
+ */
+async function loadTooltips(statsURL, blocks) {
+    // Fetch the latest stats report
+    const req = new XMLHttpRequest();
+    req.onload = (event) => {
+        const stats = JSON.parse(req.response);
+        // Assign data to each block
+        for (block of blocks) {
+            const tooltip = buildTooltip(stats[block.id]);
+            block.appendChild(tooltip);
+        }
+    };
+    req.open("GET", statsURL, true);
+    req.send();
+}
+/*
+
+/**
+ * Build a tooltip HTML element.
+ *
+ * Currently always builds the EDN elements. Would like it to take JSON for the
+ * parameters.
+ *
+ * @param stats Object of stats for this block with these properties:
+ *     * name
+ *     * version
+ *     * design_stage (may be null)
+ *     * verification_stage (may be null)
+ *     * total_runs (may be null)
+ *     * total_passing (may be null)
+ * @return A HTMLElement tooltip built from the given stats.
```

Comment:
Nit:

For JSDoc style docstrings, I don't think there should be anything before the type.

```suggestion
 * @return {HTMLElement} A tooltip element built from the given stats.
```

Also it [looks like](https://jsdoc.app/tags-returns.html) the type should be surrounded by curly braces.

This applies to [parameters](https://jsdoc.app/tags-param.html) as well.

Reply:
Sure, I'll update. I wasn't actually aware of JSDoc, just writing it in a doxygen style

Reply:
Fixed.